tests/drivers/random/randomnumgen_basic_api/src/main.c

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2017 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <random.h>
+#include <ztest.h>
+
+/*
+ * @addtogroup t_randomnumgen_basic_api
+ * @{
+ * @defgroup t_random_get_entropy test_random_get_entropy
+ * @brief TestPurpose: verify Random number generator works
+ * @details
+ * - Test Steps
+ *   -# Read random numbers from RNG driver.
+ *   -# Verify whether buffer overflow occured or not.
+ *   -# Verify whether buffer completly filled or not.
+ * - Expected Results
+ *   -# Random number should be generated.
+ * @}
+ */
+
+#define BUFFER_LENGTH 10
+
+static int recheck_random_entropy(struct device *dev, char *buffer, char num)
```

Comment:
This function can be called `check_random_entropy()`, and be called twice from `test_random_get_entropy()`, with a different `num` parameter.